src/client/sky.cpp

```diff
@@ -766,3 +624,87 @@ void Sky::update(float time_of_day, float time_brightness,
+	std::array<video::S3DVertex, 4> vertices;
+	if (!m_sun_texture) {
+		driver->setMaterial(m_materials[1]);
+		const float sunsizes[4] = {sunsize * 1.7f, sunsize * 1.2f, sunsize, sunsize * 0.7f};
+		video::SColor c1 = suncolor;
+		video::SColor c2 = suncolor;
+		c1.setAlpha(0.05 * 255);
+		c2.setAlpha(0.15 * 255);
+		const video::SColor colors[4] = {c1, c2, suncolor, suncolor2};
+		for (int i = 0; i < 4; i++) {
+			vertices = draw_sky_body(-sunsizes[i], sunsizes[i], colors[i]);
+			vertices = place_sky_body(vertices, 90, wicked_time_of_day * 360 - 90);
+			driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+		}
+	} else {
+		driver->setMaterial(m_materials[3]);
+		float d = sunsize * 1.7;
+		video::SColor c;
+		if (m_sun_tonemap)
+			c = video::SColor (0, 0, 0, 0);
+		else
+			c = video::SColor (255, 255, 255, 255);
+		vertices = draw_sky_body(-d, d, c);
+		vertices = place_sky_body(vertices, 90, wicked_time_of_day * 360 - 90);
+		driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+	}
+}
+
+
+void Sky::draw_moon(video::IVideoDriver *driver, float moonsize, video::SColor mooncolor, video::SColor mooncolor2, float wicked_time_of_day) {
+	static const u16 indices[4] = {0, 1, 2, 3};
+	std::array<video::S3DVertex, 4> vertices;
+	if (!m_moon_texture) {
+		driver->setMaterial(m_materials[1]);
+		const float moonsizes_1[4] = {-moonsize * 1.9f, -moonsize * 1.3f, -moonsize, -moonsize};
+		const float moonsizes_2[4] = {moonsize * 1.9f, moonsize * 1.3f, moonsize, moonsize * 0.6f};
+		video::SColor c1 = mooncolor;
+		video::SColor c2 = mooncolor;
+		c1.setAlpha(0.05 * 255);
+		c2.setAlpha(0.15 * 255);
+		const video::SColor colors[4] = {c1, c2, mooncolor, mooncolor2};
+		for (int i = 0; i<4; i++) {
+			vertices = draw_sky_body(moonsizes_1[i], moonsizes_2[i], colors[i]);
+			vertices = place_sky_body(vertices, -90, wicked_time_of_day * 360 - 90);
+			driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+		}
+	} else {
+		driver->setMaterial(m_materials[4]);
+		float d = moonsize * 1.9;
+		video::SColor c;
+		if (m_moon_tonemap)
+			c = video::SColor (0, 0, 0, 0);
+		else
+			c = video::SColor (255, 255, 255, 255);
+		vertices = draw_sky_body(-d, d, c);
+		vertices = place_sky_body(vertices, -90, wicked_time_of_day * 360 - 90);
+		driver->drawIndexedTriangleFan(&vertices[0], 4, indices, 2);
+	}
+}
+
+
+std::array<video::S3DVertex, 4> Sky::draw_sky_body(float pos_1, float pos_2, video::SColor c) {
+	const f32 t = 1.0f;
+	const f32 o = 0.0f;
+	std::array<video::S3DVertex, 4> vertices;
+	vertices[0] = video::S3DVertex(pos_1, pos_1, -1, 0, 0, 1, c, t, t);
+	vertices[1] = video::S3DVertex(pos_2, pos_1, -1, 0, 0, 1, c, o, t);
+	vertices[2] = video::S3DVertex(pos_2, pos_2, -1, 0, 0, 1, c, o, o);
+	vertices[3] = video::S3DVertex(pos_1, pos_2, -1, 0, 0, 1, c, t, o);
+	return vertices;
+}
+
+
+std::array<video::S3DVertex, 4> Sky::place_sky_body(std::array<video::S3DVertex, 4> vertices, float horizon_position, float day_position) {
```

Comment:
Please wrap at 90 columns (with tab size 4) and place the opening curly bracket at the start of a new line.